### PR TITLE
Windows gstreamer port

### PR DESCRIPTION
**Pull request by dtmoodie**
Gstreamer distributes windows binaries, it was fairly trivial to make it possible to modify the Linux gstreamer code to work in windows as well.  Only tested on MSVC.

### CODE REVIEW

**Comment by alalek** [comment]
OpenCV modules should not contain non-BSD compatible source code.

**Comment by vpisarev** [comment]
@dtmoodie, thanks for the contribution! I shall add to @alalek's comment that if/when you eliminate GPL code (which we cannot accept indeed, because it will make OpenCV a GPL library instead of BSD library) from your patch, please also move the headers from modules/videio/include/opencv2/winport to modules/videoio/src, because they are not a part of our exposed API, they are completely about implementation.

**Comment by dtmoodie** [comment]
Hello!
After some testing it looks like I can completely remove the non-BSD code because the needed functions are either included elsewhere or not used with the windows port.

**Comment by alalek** [comment]
It works!

1) Please replace tab symbols with spaces
2) Please squash commits into one (to eliminate unnecessary intermediate changes from history, like removed files and unused/removed GPL code)
Something like this should work (there are merge commits, so little help of GitHub API is used here):
```
# push your current changes to GitHub
git remote add upstream https://github.com/Itseez/opencv.git
git fetch upstream
git fetch upstream +pull/5723/merge:pr5723_m
git checkout -B win_gst upstream/master
git checkout pr5723_m -- .
git commit -am "Gstreamer 1.0 on windows."
git push -f origin win_gst
```

File: cmake/OpenCVFindLibsVideo.cmake [hunk 1]
```diff
@@ -14,22 +14,35 @@ endif(WITH_VFW)
...
-  CHECK_MODULE(gstreamer-app-1.0 HAVE_GSTREAMER_APP)
-  CHECK_MODULE(gstreamer-riff-1.0 HAVE_GSTREAMER_RIFF)
-  CHECK_MODULE(gstreamer-pbutils-1.0 HAVE_GSTREAMER_PBUTILS)
+  if(WIN32)
+    SET(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH};${CMAKE_CURRENT_LIST_DIR}")
+    FIND_PACKAGE(GstreamerWindows)
+	IF(GSTREAMER_gstbase_LIBRARY AND GSTREAMER_gstvideo_LIBRARY AND GSTREAMER_gstapp_LIBRARY AND GSTREAMER_gstpbutils_LIBRARY AND GSTREAMER_gstriff_LIBRARY)
+	  set(HAVE_GSTREAMER TRUE)
+	  set(GSTREAMER_BASE_VERSION 1.0)
+      set(GSTREAMER_VIDEO_VERSION 1.0)
```
Comment (alalek) [inline cmake/OpenCVFindLibsVideo.cmake]: 3 tabs on these lines

File: modules/videoio/CMakeLists.txt [hunk 2]
```diff
@@ -92,6 +92,10 @@ if(HAVE_DC1394)
 endif(HAVE_DC1394)
 
 if(HAVE_GSTREAMER)
+  IF(WIN32)
+    INCLUDE_DIRECTORIES(${GSTREAMER_INCLUDE_DIR})
+	list(APPEND VIDEOIO_LIBRARIES ${GSTREAMER_LIBRARIES})
```
Comment (alalek) [inline modules/videoio/CMakeLists.txt]: 1 tab is here

**Comment by dtmoodie** [comment]
I think that should be everything you requested, thanks!  I couldn't figure out why git wouldn't squash my commits.

**Comment by alalek** [comment]
Well done! :+1:

**Pull request merged**